clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -171,11 +170,11 @@ public EcsServerClusterProvider(
       }
 
       if (clusterMap.containsKey(applicationName)) {
-        String escClusterName = StringUtils.substringBeforeLast(ecsServerGroup.getName(), "-");
+        String ecsServerClusterName = moniker.getCluster();
         boolean found = false;
 
         for (EcsServerCluster cluster : clusterMap.get(applicationName)) {
-          if (cluster.getName().equals(escClusterName)
+          if (cluster.getName().equals(ecsServerClusterName)
               && cluster.getAccountName().equals(credentials.getName())) {
```

Comment:
nit: since we're using StringUtils in most places, would you mind using StringUtils.equals here?

Reply:
Should be fixed.